### PR TITLE
[#897] - Add issue templates

### LINKED ISSUE #897: Issue templates missing
We need an issue template now that we have GitHub issues open.
This could get us started:
https://gist.github.com/bjankord/0db8add11881ba4ba8003ece82561f76

We should include

- [ ] here is not the place to report security issues
- [ ] expected / actual behaviour (if it is a bug)
- [ ] ...?

### PR DESCRIPTION
**Pull request by EdgarRamirezFuentes**
Resolve #897 by adding bug report, documentation related, enhancement request, feature request, and question issue templates.

Following this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [GitHub issue](https://github.com/apache/shiro/issues) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a GitHub issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[#XXX] - Fixes bug in SessionManager`,
       where you replace `#XXX` with the appropriate GitHub issue. Best practice
       is to use the GitHub issue title in the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] add `fixes #XXX` if merging the PR should close a related issue.
 - [ ] Run `mvn clean install apache-rat:check` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If you have a group of commits related to the same change, please squash your commits into one and force push your branch using `git rebase -i`. 
 
Trivial changes like typos do not require a GitHub issue (javadoc, comments...). 
In this case, just format the pull request title like `[DOC] - Add javadoc in SessionManager`.
 
If this is your first contribution, you have to read the [Contribution Guidelines](https://github.com/apache/shiro/blob/master/CONTRIBUTING.md)

If your pull request is about ~20 lines of code you don't need to sign an [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) 
if you are unsure please ask on the developers list.

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Review by lprimak — APPROVED** [review]
Im inclined to approve this.
However, since I don't really have any expertise in setting these up,
can somebody else take a look as well :)

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        - Linux
+        - Other
+    validations:
+      required: true
+
+  - type: dropdown
+    attributes:
+      label: Shiro version
+      description: Describe the Shiro version.
+      options:
```
Comment (fpapon) [inline .github/ISSUE_TEMPLATE/bug_report.yml:56]: I'm not sure this is a good idea to provide a version list because we will need to maintain it up to date.
May be a simple input text field should be enough.

**Review by fpapon — CHANGES_REQUESTED** [review]
Just a little comment about the version list in the bug report template, otherwise LGTM.
@EdgarRamirezFuentes Thank you for your contribution!

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (EdgarRamirezFuentes) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Understood! 
I made the changes in the bug report file, and switched the Shiro version to a textarea. 😄

**Review by fpapon — APPROVED** [review]
LGTM!

**Comment by fpapon** [comment]
@bmarwell is it ok with what you had in mind?

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        first to see whether the same issue was reported already.
+      options:
+        - label: >
+            I had searched in the [issues](https://github.com/apache/shiro/issues?q=is%3Aissue) and found
+            no similar issues.
+          required: true
+
+  - type: dropdown
+    attributes:
+      label: Environment
```
Comment (bmarwell) [inline .github/ISSUE_TEMPLATE/bug_report.yml:42]: I think we can omit the environment b/c Java.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      label: Shiro version
+      description: Describe the Shiro version.
+      placeholder: >
+        Describe the Shiro version here.
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: What happened
```
Comment (bmarwell) [inline .github/ISSUE_TEMPLATE/bug_report.yml:63]: "what was the actual outcome"

Comment (bmarwell) [reply .github/ISSUE_TEMPLATE/bug_report.yml:63]: It's missing "what did you expect instead"

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (bmarwell) [reply .github/ISSUE_TEMPLATE/bug_report.yml:42]: Maybe replace with a text field for application server, Java version, etc

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      label: Debug logs
+      description: Anything else we need to know?
+      placeholder: >
+        Add your debug logs here.
+    validations:
+      required: false
+
+  - type: checkboxes
+    attributes:
+      label: Are you willing to submit PR?
```
Comment (bmarwell) [inline .github/ISSUE_TEMPLATE/bug_report.yml:92]: This is not required imho. Or maybe rephrase it? @bdemers might want to suggest something?

**Review by bmarwell — CHANGES_REQUESTED** [review]

**Comment by bmarwell** [comment]
> @bmarwell is it ok with what you had in mind?

It's a great start!

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (lprimak) [reply .github/ISSUE_TEMPLATE/bug_report.yml:42]: i.e Environment? :)

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 3 above)
Comment (EdgarRamirezFuentes) [reply .github/ISSUE_TEMPLATE/bug_report.yml:63]: So, I change the label to "what was the actual outcome" instead of "What happened", and add a new textarea with the label "what did you expect instead"?

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (EdgarRamirezFuentes) [reply .github/ISSUE_TEMPLATE/bug_report.yml:42]: You mean something like this?
Captura de pantalla 2023-07-07 a la(s) 16 15 56

**Comment by lprimak** [comment]
I would say to merge this "as is" and then update it when we figure out what the exact copy should read.
WDYT?

**Comment by EdgarRamirezFuentes** [comment]
@lprimak 
I could commit the modifications I made in the environment field, and It could be merged and modified later as you mentioned.

**Comment by lprimak** [comment]
@bmarwell What do you think?

**Comment by bmarwell** [comment]
Yes, please change accordingly.

**Comment by EdgarRamirezFuentes** [comment]
Done!
I modified the environment field to be a text field, and added the fields to explain what was the actual and expected outcome.

**Review by bmarwell — APPROVED** [review]
Yup, that's a start!

**Comment by fpapon** [comment]
@EdgarRamirezFuentes thanks for your first contribution!
